app/controllers/name_controller/create_and_edit_name.rb

```diff
@@ -139,32 +143,52 @@ def perform_merge_names(new_name)
       @name, new_name = new_name, @name
       old_display_name_for_log = @name[:display_name]
     end
+    @name.display_name = old_display_name_for_log
+
     # Fill in author if other has one.
     if new_name.author.blank? && @parse.author.present?
       new_name.change_author(@parse.author)
     end
     new_name.change_deprecated(change_deprecated) unless change_deprecated.nil?
-    @name.display_name = old_display_name_for_log
+
+    # move associations from @name to new_name and destroy @name
     new_name.merge(@name)
     args = { this: @name.real_search_name, that: new_name.real_search_name }
     flash_notice(:runtime_edit_name_merge_success.t(args))
+
     @name = new_name
+    if new_name.icn_id != old_identifier
+      email_admin_name_change(old_identifier: old_identifier)
+    end
     @name.save
   end
 
   def set_name_author_and_rank
     return unless name_unlocked?
 
-    email_admin_name_change unless minor_change? || just_adding_author?
+    unless minor_change?
+      email_admin_name_change(
+        old_identifier: Name.find(@name.id).icn_id
```

Comment:
I feel like there should be a better way to do this than querying the database.  Doesn't ActiveRecord give access to what the old state of a parameter is up until the change is saved?  Maybe something like @name.icn_id_was or something like that?

Also, I feel like ICN id is somewhat equivalent to author in this sense.  If the user is just adding an ICN id, I don't see that as worth an admin notification, the same way adding an author doesn't warrant a notification.  (The logic is that I want to be made aware of potentially damaging changes, but adding data where there was none before -- adding an author to a name which had no author, or adding an ICN id to a name without one --  is nondestructive, so I don't care about keeping an eye open for potential abuse there.)

So I would personally recommend reverting this and doing it more like it was before:

email_admin_name_change unless minor_change? || just_adding_author_or_icn_id?

If that can be done.

Reply:
Sorry, I see that minor_change? now includes just_adding_author?, and that probably makes sense.

Reply:
1. I'll look into how to avoid hitting the db. It did bother me that I was doing this, but I was too lazy/tired to figure out how to avoid it.
2. I won't notify admin if the only change is filling in a blank ICN id